### ISSUE TITLE
Timeout during synchronize

### ISSUE THREAD

**Issue by hackartisan** [pulibrary/abid#121]
User attempted to synchronize a batch of 150 containers. It takes several seconds and then a 405 nginx error appears.

Here are the application log entries:
```
1757867:I, [2023-04-26T15:10:43.690931 #31960]  INFO -- : [7ded2951-bf50-4cec-9adc-72b42a7e597f] Started POST "/batches/438/synchronize" for 10.8.59.223 at 2023-04-26 15:10:43 +0000
1757868-I, [2023-04-26T15:10:43.691969 #31960]  INFO -- : [7ded2951-bf50-4cec-9adc-72b42a7e597f] Processing by BatchesController#synchronize as HTML
1757869-I, [2023-04-26T15:10:43.692020 #31960]  INFO -- : [7ded2951-bf50-4cec-9adc-72b42a7e597f]   Parameters: {"authenticity_token"=>"[FILTERED]", "id"=>"438"}
--
1758947:I, [2023-04-26T15:28:41.853149 #31960]  INFO -- : [f6a33eb9-dfbe-45f0-88a4-3d03d51ca8c8] Started POST "/batches/438/synchronize" for 10.8.59.223 at 2023-04-26 15:28:41 +0000
1758948-I, [2023-04-26T15:28:41.854151 #31960]  INFO -- : [f6a33eb9-dfbe-45f0-88a4-3d03d51ca8c8] Processing by BatchesController#synchronize as HTML
1758949-I, [2023-04-26T15:28:41.854206 #31960]  INFO -- : [f6a33eb9-dfbe-45f0-88a4-3d03d51ca8c8]   Parameters: {"authenticity_token"=>"[FILTERED]", "id"=>"438"}
--
1759288:I, [2023-04-26T15:34:20.885287 #31960]  INFO -- : [d6a5a2bd-3262-4464-9cc6-7e0c835e0da2] Started POST "/batches/438/synchronize" for 172.20.217.192 at 2023-04-26 15:34:20 +0000
1759289-I, [2023-04-26T15:34:20.886237 #31960]  INFO -- : [d6a5a2bd-3262-4464-9cc6-7e0c835e0da2] Processing by BatchesController#synchronize as HTML
1759290-I, [2023-04-26T15:34:20.886295 #31960]  INFO -- : [d6a5a2bd-3262-4464-9cc6-7e0c835e0da2]   Parameters: {"authenticity_token"=>"[FILTERED]", "id"=>"438"}
--
1842831:I, [2023-04-27T14:49:04.011251 #31960]  INFO -- : [207cc1f6-8457-40a3-9ea3-e84b4b73cd9f] Started POST "/batches/438/synchronize" for 98.115.95.174 at 2023-04-27 14:49:04 +0000
1842832-I, [2023-04-27T14:49:04.012475 #31960]  INFO -- : [207cc1f6-8457-40a3-9ea3-e84b4b73cd9f] Processing by BatchesController#synchronize as HTML
1842833-I, [2023-04-27T14:49:04.012536 #31960]  INFO -- : [207cc1f6-8457-40a3-9ea3-e84b4b73cd9f]   Parameters: {"authenticity_token"=>"[FILTERED]", "id"=>"438"}
```

Here's a datadog log entry, from lib-acd2:
```
[ip-redacted] - - [27/Apr/2023:14:50:04 +0000] "POST /batches/438/synchronize HTTP/2.0" 405 559 "https://abid.princeton.edu/batches?created_batch=438" [ip-redacted]:80, 504, 60.001, 60.001"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36" "-" hostname="abid.princeton.edu"
2023/04/26 15:35:20 [error] 957173#957173: *29671284 upstream timed out (110: Connection timed out) while reading response header from upstream, client: [ip-redacted], server: abid.princeton.edu, request: "POST /batches/438/synchronize HTTP/2.0", upstream: "http://128.112.200.198:80/batches/438/synchronize", host: "abid.princeton.edu", referrer: "https://abid.princeton.edu/batches?created_batch=438"
```

## Sudden priority justification
Blocking staff workflow

**Comment by hackartisan** [comment]
In looking at the data I found that the batch was successfully synchronized. So while we still don't know what happened here exactly, it's no longer blocking work.

**Comment by hackartisan** [comment]
related to https://github.com/pulibrary/princeton_ansible/issues/2603?

**Comment by regineheberlein** [comment]
This may or may not be relevant, but I thought it worth mentioning here just in case:

I've looked into other reports of lag in ASpace and newly-created objects not becoming available in the UI since yesterday. I went in thinking it's an indexing thing, possibly related to bulk updates on Agents I've been running this week. However, I now think this issue is more specific to top_containers.

Here's how I'm arriving at this suspicion:

- Lyrasis checked the ASpace and SOLR servers for us and the logs, and Blake says there's nothing in there to indicate that indexing is or recently was slow. 
- I've followed up on all the reports of disappearing objects in the UI, and the only ones that have stuck (i.e. that I can reproduce, and that I can't exclude as unrelated or mere oversight) are top_containers.
- Staff reports of sluggish performance came in after Lynn had reported the AbID 405

So I think whatever is going on has to do with top_containers. I'll ask Blake to take a closer look with that angle. If I learn anything more, I'll post it here.
